inspirehep/utils/stats.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+import operator
+
+
+class Stats(object):
+
+    """Stats is a class to calculate common metrics"""
+
+    @staticmethod
+    def calculate_hindex(*args, **kwargs):
+        """
+        Given a citation dictionary, in the format {recid: citation_count}
+        returns the hindex.
+        :param args:
+        :param kwargs:
+        :return: h index for a given list of citations.
+        """
+        _record_citations = kwargs.pop('citations')
```

Comment:
Note that if the parameter `citations` is not mandatory this line will give `KeyError`. So you can either add a default value to the `pop` function or make `citations` mandatory by adding it to the `calculate_hindex` parameters.